gapic-common/lib/gapic/common/polling_harness.rb

```diff
@@ -60,6 +60,7 @@ def wait
             raise unless retry_policy.retry_error? e
           end
           retry_policy.perform_delay!
+          return response unless retry_policy.retry_with_deadline?
```

Comment:
I suspect you're going to get an undefined local variable here because `response` is not in scope; it exists only within the `begin`-`end` block above. What exactly do we want to return if we time out? Should we return nil? Should we be raising something?

Reply:
I have decided to go with `nil` as I think it would be simpler for client usage. I also added some docstring for it